lib/autoupdate.ps1

```diff
@@ -281,44 +281,141 @@ function get_hash_for_app([String] $app, $config, [String] $version, [String] $u
     return $hash
 }
 
-function update_manifest_with_new_version($json, [String] $version, [String] $url, [String] $hash, $architecture = $null) {
-    $json.version = $version
-
-    if ($null -eq $architecture) {
-        if ($json.url -is [System.Array]) {
-            $json.url[0] = $url
-            $json.hash[0] = $hash
-        } else {
-            $json.url = $url
-            $json.hash = $hash
+function Update-ManifestProperty {
+    [CmdletBinding(SupportsShouldProcess = $true)]
+    [OutputType([Boolean])]
+    param (
+        [Parameter(Mandatory = $true, Position = 1)]
+        [PSObject]
+        $Manifest,
+        [Parameter(ValueFromPipeline = $true, Position = 2)]
+        [String[]]
+        $Property,
+        [String]
+        $AppName,
+        [String]
+        $Version,
+        [Alias("Matches")]
+        [HashTable]
+        $Substitutions
+    )
+    begin {
+        function PropertyHelper {
```

Comment:
Is it useful to redeclare these every time when `Update-ManifestProperty` is called?

Reply:
Wow, if used manually, there is no difference, but for batch operation, moving them outside should be better.